### PR TITLE
[Locale ][Api] Get locale refactor api

### PR DESCRIPTION
**Pull request by AdamKasp**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no/
| Related tickets | based #12766
| License         | MIT

### CODE REVIEW

File: config/packages/lexik_jwt_authentication.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 lexik_jwt_authentication:
-    secret_key: '%env(resolve:JWT_SECRET_KEY)%'
-    public_key: '%env(resolve:JWT_PUBLIC_KEY)%'
-    pass_phrase: '%env(JWT_PASSPHRASE)%'
+    private_key_path: '%kernel.project_dir%/config/jwt/private-test.pem'
+    public_key_path: '%kernel.project_dir%/config/jwt/public-test.pem'
+    pass_phrase: 'adam'
```
Comment (stloyd) [inline config/packages/lexik_jwt_authentication.yaml:4]: Really? 😄

Comment (AdamKasp) [reply config/packages/lexik_jwt_authentication.yaml]: :D

File: features/locale/getting_available_locales_in_current_channel.feature [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        Then I should have 2 locales
+        And the "English (United States)" locale with code "en_US" should be available
+        And the "Polish (Poland)" locale with code "pl_PL" should be available
+        But the "German (Germany)" locale with code "de_DE" should not be available
+
+    @api
+    Scenario: Get locales details
+        When I get "English (United States)" locale
+        Then I should have "English (United States)" with code "en_US"
+
```
Comment (Tomanhez) [inline features/locale/getting_available_locales_in_current_channel.feature:25]: redundant blank line

File: features/locale/managing_locales/adding_locale.feature [hunk 3]
```diff
@@ -7,7 +7,7 @@ Feature: Adding a new locale
     Background:
         Given I am logged in as an administrator
 
-    @ui @api
```
Comment (Tomanhez) [inline features/locale/managing_locales/adding_locale.feature:10]: Why UI is disabled?

File: src/Sylius/Behat/Resources/config/suites/api/locale/locales.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+                - sylius.behat.context.transform.channel
+                - sylius.behat.context.transform.locale
+                - sylius.behat.context.transform.shared_storage
+                - sylius.behat.context.transform.lexical
+
+                - sylius.behat.context.setup.channel
+                - sylius.behat.context.setup.locale
+
+                - sylius.behat.context.api.shop.locale
+            filters:
```
Comment (arti0090) [inline src/Sylius/Behat/Resources/config/suites/api/locale/locales.yaml:19]:
```suggestion
                - sylius.behat.context.api.shop.locale
                
            filters:
```

File: tests/Api/Admin/LocalesTest.php [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+namespace Sylius\Tests\Api\Admin;
+
+use Sylius\Tests\Api\JsonApiTestCase;
+use Symfony\Component\HttpFoundation\Response;
+
+final class LocalesTest extends JsonApiTestCase
+{
+    /** @test */
+    public function it_gets_locales(): void
```
Comment (arti0090) [inline tests/Api/Admin/LocalesTest.php:22]:
```suggestion
    public function it_gets_locales_as_loged_in_administrator(): void
```
or something like this, and you should also make a test for unathorized (I assume) :slightly_smiling_face:

**Review by arti0090 — APPROVED** [review]

File: tests/Api/Responses/Expected/shop/get_locale_response.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{
+  "@context":"\/api\/v2\/contexts\/Locale",
+  "@id":"\/api\/v2\/shop\/locales\/en_US",
+  "@type":"Locale",
+  "code":"en_US",
+  "name":"English (United States)"
```
Comment (GSadee) [inline tests/Api/Responses/Expected/shop/get_locale_response.json:6]: Use 4 spaces instead of 2

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Adam! :1st_place_medal:

**Pull request merged**